### PR TITLE
sign-file: Add SRK REVOKE option

### PR DESCRIPTION
**Pull request by angolini**
Signed-off-by: Daiane Angolini <daiane.angolini@foundries.io>

### CODE REVIEW

**Comment by angolini** [comment]
I'm going to propose something for AHAB as well

**Comment by angolini** [comment]
I tested this PR only for imx6ullevk-sec
the AHAB procedure should be the same, but I have *not* tested it personally

File: security/imx_ahab/sign-file.sh [hunk 1]
```diff
@@ -87,6 +92,7 @@ if [ "${DISPLAY_USAGE}" = "1" ]; then
     echo "  --spl: SPL binary to sign             [--spl is required]"
     echo "  --key-dir: location for key files     [default: ${KEY_DIR}]"
     echo "  --srk-index: the key to sign with     [default: 1]"
+    echo "  --enable-revoke: used only with the intent to unlock the revokation       [default: ${ENABLE_REVOKE}]"
```
Comment (ricardosalveti) [inline security/imx_ahab/sign-file.sh:95]: Maybe 'unlock key revocation'?

File: security/imx_ahab/sign-file.sh [hunk 2]
```diff
@@ -122,6 +128,14 @@ fi
 # working file used for signature
 cp ${WORK_FILE} ${WORK_FILE}.mod
 
+# enable revokation
```
Comment (ricardosalveti) [inline security/imx_ahab/sign-file.sh:131]: Revocation?

File: security/imx_hab4/sign-file.sh [hunk 3]
```diff
@@ -107,6 +112,7 @@ if [ "${DISPLAY_USAGE}" = "1" ]; then
     echo "  --fix-sdp-dcd: turn on clear / restore DCD addr for SPD SPL binary"
     echo "  --srk-index: the key to sign with     [default: 1]"
     echo "  --engine: the the engine to use       [default: CAAM]"
+    echo "  --enable-revoke: used only with the intent to unlock the revokation       [default: ${ENABLE_REVOKE}]"
```
Comment (ricardosalveti) [inline security/imx_hab4/sign-file.sh:115]: Same.

File: security/imx_hab4/sign-file.sh [hunk 4]
```diff
@@ -264,6 +270,14 @@ echo "Verification index = 2" >> ${CSF_TEMPLATE}.csf-config
 # use .mod file in case we cleared DCD info
 echo "Blocks = ${HAB_BLOCKS} \"${WORK_FILE}.mod\"" >> ${CSF_TEMPLATE}.csf-config
 
+# enable revokation
```
Comment (ricardosalveti) [inline security/imx_hab4/sign-file.sh:273]: Same.

File: security/imx_ahab/sign-file.sh [hunk 5]
```diff
@@ -122,6 +128,14 @@ fi
...
+# info from CST_UG, section “B SRK revocation on HABv4”, page 18
```
Comment (vanmaegima) [inline security/imx_ahab/sign-file.sh:132]: not sure I would like to mention all these details, maybe just the section name is enough (if AN gets updated in the future)
```suggestion
# info from CST_UG, section “B SRK revocation on HABv4”, page 18
```

**Comment by angolini** [comment]
@vanmaegima @ricardosalveti , updated

**Review by vanmaegima — APPROVED** [review]
lgtm

**Review by Tim-Anderson — APPROVED** [review]
LGTM

**Pull request merged**